internal/pkg/runtime/engines/singularity/rpc/server/server_linux.go

```diff
@@ -33,6 +38,52 @@ func (t *Methods) Mount(arguments *args.MountArgs, reply *int) (err error) {
 	return err
 }
 
+// Crypt decrypts the loop device
+func (t *Methods) Decrypt(arguments *args.CryptArgs, reply *string) (err error) {
+
+	sylog.Debugf("In Crypt RPC")
+	sylog.Debugf("Crypt RPC running in PID %d", os.Getpid())
+	sylog.Debugf("Loop device is %s", arguments.Loopdev)
+
+	fmt.Print("Enter the password to decrypt File System: ")
+	password, err := terminal.ReadPassword(int(syscall.Stdin))
```

Comment:
Any specific reason to do that instead of reading here?

Reply:
Well actually this won't work when running a Singularity instance because there is no terminal available at this stage, this is also the case in master process but that could be fixed more easily for master process in the future. Additionally I'm not comfortable to let user interact with a program executed with root privileges

Reply:
I created #3851 to supply encryption keys non-interactively.